### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by hanzei**
Downstream from https://github.com/mattermost/mattermost-plugin-nps/pull/5

Credits to @hmhealey for creating this file!

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+[webapp/package.json]
+indent_size = 2
+
+[Makefile,*.mk]
+indent_style = tab
+
+[*.md]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = false
```
Comment (lieut-data) [inline .editorconfig]: Is there a specific reason this is `false`?

Comment (hanzei) [reply .editorconfig]: You can do linebreaks in `md` by ending a line with two spaces. Some people use this and we might want to support it.

Comment (lieut-data) [reply .editorconfig]: Fascinating, I had no idea! Seems both useful and evil...

Comment (hanzei) [reply .editorconfig]: I'm not 100% sure that this is a good part of markdown...

**Pull request merged**